### PR TITLE
8280026: Cleanup of IGV printing

### PR DESCRIPTION
**Pull request by neliasso**
This is a clean up of the code that controls the printing of IGV graphs. This is preparatory work for another patch.

* Default params removed when they are only used in the non-default case.
* Moved code around so that methods with similar concern are kept together
* Move some code from hpp to cpp file
* Changed name of print_method with actual implementation to print_method_impl to clarify intent
* Removed the print_method that was used with a node idx. I once introduced it for debugging - but now there are also other versions that pass a node.

Please revew,
Nils

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8280026](https://bugs.openjdk.java.net/browse/JDK-8280026): Cleanup of IGV printing

### Reviewers
 * [Christian Hagedorn](https://openjdk.java.net/census#chagedorn) (@chhagedorn - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7088/head:pull/7088` \
`$ git checkout pull/7088`

Update a local copy of the PR: \
`$ git checkout pull/7088` \
`$ git pull https://git.openjdk.java.net/jdk pull/7088/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7088`

View PR using the GUI difftool: \
`$ git pr show -t 7088`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7088.diff">https://git.openjdk.java.net/jdk/pull/7088.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/opto/compile.cpp [hunk 1]
```diff
@@ -4773,43 +4789,46 @@ void Compile::print_method(CompilerPhaseType cpt, const char *name, int level) {
...
 }
 
+bool Compile::should_print(int level) {
+#ifndef PRODUCT
+  if (PrintIdealGraphLevel < 0) { // disabled by the user
+    return false;
+  }
+
+  bool need = directive()->IGVPrintLevelOption >= level;
+  if (need && !_printer) {
```
Comment (chhagedorn) [inline src/hotspot/share/opto/compile.cpp:4823]: Could be changed into `_printer == NULL`.

**Review by chhagedorn — APPROVED** [review]
Otherwise, looks good!

**Comment by neliasso** [comment]
Thanks for the review @chhagedorn !

**Review by TobiHartmann — APPROVED** [review]
Looks good.

**Pull request closed without merging**